### PR TITLE
M3-4582 Change: Use a shared component for view-only permissions tables.

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

For OBJ access tokens and API access tokens, instead of showing a disabled version of the scopes form (with a bunch of radio buttons), if a user is in view-only mode instead use check marks for selected permissions. I opted to go with a shared (new) component for this, though it isn't clear if it's worth it in this case.

Screen Shot 2020-12-01 at 4 04 07 PM

@ndukatz there are a few existing e2e tests that used data-qa attributes that are lost using this approach (nothing in Cypress though). Please let me know if I broke anything you're working on.

@tiffwong making this change meant that the aria stuff needed to be dynamic. The results are a bit different but I think they're ok. Please let me know.

- [ ] unit tests

### CODE REVIEW

File: packages/manager/src/features/Profile/APITokens/APITokenMenu.tsx [hunk 1]
```diff
@@ -26,25 +26,25 @@ class APITokenMenu extends React.Component<CombinedProps> {
     } = this.props;
 
     return (closeMenu: Function): Action[] => {
-      const actions = !isThirdPartyAccessToken
+      return !isThirdPartyAccessToken
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/APITokens/APITokenMenu.tsx:29]: These are linter fixes

File: packages/manager/src/features/Profile/APITokens/APITokenDrawer.tsx [hunk 2]
```diff
@@ -308,51 +309,41 @@ export class APITokenDrawer extends React.Component<CombinedProps, State> {
...
                   className={classes.noneCell}
                 >
-                  <Radio
-                    name={scopeTup[0]}
-                    disabled={mode !== 'create' && scopeTup[1] !== 0}
-                    checked={scopeTup[1] === 0}
-                    value="0"
+                  <AccessCell
+                    active={scopeTup[1] === 0}
+                    scope="2"
```
Comment (johnwcallahan) [inline packages/manager/src/features/Profile/APITokens/APITokenDrawer.tsx:314]: Think this should be "0"... currently there's some incorrect behavior when you click "None" for a single scope.

**Review by johnwcallahan — CHANGES_REQUESTED** [review]
UI looks good, one bug in the PAT drawer.

Not sure about the accessibility here. Maybe the span containing the SVG needs something like:

```
        role="checkbox"
        aria-checked="true"
        aria-disabled="true"
```

**Comment by Jskobos** [comment]
> UI looks good, one bug in the PAT drawer.
> 
> Not sure about the accessibility here. Maybe the span containing the SVG needs something like:
> 
> ```
>         role="checkbox"
>         aria-checked="true"
>         aria-disabled="true"
> ```

I'm not sure about accessibility either, I feel weird telling people it's a checkbox when it isn't. I added an aria-label to the span instead, let me know what you think. @tiffwong please weigh in as well.

**Review by johnwcallahan — APPROVED** [review]
The ARIA label makes sense to me.

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessCell.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const classes = useStyles();
+
+  if (viewOnly) {
+    if (!active) {
+      return null;
+    }
+    return (
+      <span
+        className={classes.checkIcon}
+        aria-label={`This token has ${scope} access for ${scopeDisplay}`}
```
Comment (tiffwong) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessCell.tsx:42]: Currently, the `aria-label` isn't accessible so this never gets read but can easily be fixed by adding `tabIndex={0}`

**Review by tiffwong — CHANGES_REQUESTED** [review]
This looks great! The check icons are a great visual touch to prevent users from thinking they can edit the permissions 💯 

The only comment I have is to add a `tabIndex` so that users can access the `aria-label` and I agree with not labeling it as a checkbox since that could confuse the user into thinking they can edit it (which defeats the whole purpose of this)

**Review by tiffwong — APPROVED** [review]

**Pull request merged**